src/sync/nightly_tests.rs

```diff
@@ -12,3 +12,17 @@ fn async_closure() {
     assert_eq!(gen.resume(), GeneratorState::Yielded(10));
     assert_eq!(gen.resume(), GeneratorState::Complete("done"));
 }
+
+#[test]
+fn sync_proc_macro_closure() {
+    #[yielder_cls(u8)]
+    let gen = unsafe {
```

Comment:
I don't think this needs to be unsafe

Reply:
The nightly tests won't work anyway they all need to be done in the test folder because the `Co` arg
is `co: ::genawaiter::stack::Co<'_, {}>`, and hopefully I can remove the closure hygiene problem, I will remove all the nightly proc_macro tests for now then move them into the test folder when I can.

Reply:
Fun fact, the feature="nightly" tests don't even run in CI. I just added them to sanity check one of the examples. I should probably either add it to CI or delete that code altogether.

For now I won't touch it so we don't get merge conflicts—do whatever makes sense to you.